src/loader/mod.rs

```diff
@@ -378,6 +395,117 @@ impl KernelLoader for BzImage {
     }
 }
 
+#[cfg(feature = "arm64_pe")]
+#[cfg(target_arch = "aarch64")]
+/// Aarch64 kernel image support.
+pub struct Arm64Pe;
+
+#[cfg(feature = "arm64_pe")]
+#[cfg(target_arch = "aarch64")]
+impl KernelLoader for Arm64Pe {
+    /// Loads a Aarch64 kernel image
+    ///
+    /// Aarch64 kernel boot protocol is specified in the kernel docs
+    /// Documentation/arm/Booting and Documentation/arm64/booting.txt.
+    ///
+    /// ======aarch64 kernel header========
+    /// u32 code0;                /* Executable code */
+    /// u32 code1;                /* Executable code */
+    /// u64 text_offset;          /* Image load offset, little endian */
+    /// u64 image_size;           /* Effective Image size, little endian */
+    /// u64 flags;                /* kernel flags, little endian */
+    /// u64 res2  = 0;            /* reserved */
+    /// u64 res3  = 0;            /* reserved */
+    /// u64 res4  = 0;            /* reserved */
+    /// u32 magic = 0x644d5241;   /* Magic number, little endian, "ARM\x64" */
+    /// u32 res5;                 /* reserved (used for PE COFF offset) */
+    /// ====================================
+    ///
+    /// # Arguments
+    ///
+    /// * `guest_mem` - The guest memory region the kernel is written to.
+    /// * `kernel_start` - The offset into 'guest_mem' at which to load the kernel.
+    /// * `kernel_image` - Input vmlinux image.
+    /// * `highmem_start_address` - This is the start of the high memory, kernel should above it.
+    ///
+    /// # Returns
+    /// * KernelLoaderResult
+    fn load<F, M: GuestMemory>(
+        guest_mem: &M,
+        kernel_start: Option<GuestAddress>,
+        kernel_image: &mut F,
+        _highmem_start_address: Option<GuestAddress>,
+    ) -> Result<KernelLoaderResult>
+    where
+        F: Read + Seek,
+    {
+        const AARCH64_KERNEL_LOAD_ADDR: usize = 0x80000;
```

Comment:
These `const`s might be better placed outside the function.

Reply:
I made them trait-level `const`'s.